src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithManufacturerFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find car in DB by id: " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectAllRequest = "SELECT c.id as car_id, c.model as car_model, m.id as"
+                + " manufacturer_id, m.name as manufacturer_name, m.country as "
+                + "manufacturer_country FROM cars c JOIN manufacturers m ON "
+                + "c.manufacturer_id = m.id where c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(selectAllRequest)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all data from cars table", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarRequest = "UPDATE cars c SET model = ?, manufacturer_id = ?"
+                + " where id = ? and is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(updateCarRequest)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in DB. Car: " + car, e);
+        }
+        deleteAllRelationsBetweenCarAndDrivers(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery = "UPDATE cars SET is_deleted = true where id = ?;";
+        Optional<Car> carOptional = get(id);
+        Car car = carOptional.orElseThrow(() -> new DataProcessingException("Couldn't delete"
+                + " relations between drivers and car. ID car: " + id));
+        deleteAllRelationsBetweenCarAndDrivers(car);
+        try (Connection connection = ConnectionUtil.getConnection();

```

Comment:
```suggestion
        String deleteCarQuery = "UPDATE cars SET is_deleted = true where id = ?;";
        try (Connection connection = ConnectionUtil.getConnection();

```

we should not delete relations between deleted car and drivers hence the soft delete concept